Marlin/Marlin_main.cpp

```diff
@@ -9483,25 +9483,39 @@ inline void gcode_M999() {
 }
 
 #if ENABLED(SWITCHING_EXTRUDER)
-  inline void move_extruder_servo(uint8_t e) {
-    const int angles[2] = SWITCHING_EXTRUDER_SERVO_ANGLES;
-    MOVE_SERVO(SWITCHING_EXTRUDER_SERVO_NR, angles[e]);
-    safe_delay(500);
+  inline void move_extruder_servo(const uint8_t e) {
+    stepper.synchronize();
+    constexpr int16_t angles[] = SWITCHING_EXTRUDER_SERVO_ANGLES;
+    static_assert(COUNT(angles) / 2 >= E_STEPPERS, "SWITCHING_EXTRUDER_SERVO_ANGLES needs 2 angles per servo.");
+
+    #if EXTRUDERS & 1
```

Comment:
@thinkyhead ,
Then User is configuring Marlin, he can operate only with definition of `EXTRUDERS `in **Configuration.h**
With `SWITCHNG_EXTRUDER` we have
`EXTRUDERS `= x (user defined)
`HOTENDS `= `EXTRUDERS`
`E_STEPPERS `= `EXTRUDERS /2` (for 2 or 4 extruders) or `(EXTRUDERS + 1) / 2` (for 3 or 5 extruders)

So we’ll have the same `E_STEPPERS `for 3 and 4 `EXTRUDERS`, 
if `EXTRUDERS `= 3, we do not need to perform any switching of servo, when activating 3-rd tool (`T2`), because we have one Switching extruder(1-st Stepper for `T0 `and `T1`) and one classic extruder for `T2`
If we try to operate with `E_STEPPERS `here, Marlin will not know if it necessary to switch servo, when we activate `T2`, because it can’t recognize how many tools (`EXTRUDERS`) are defined – 3 or 4